src/main/java/com/google/devtools/build/lib/syntax/MethodDescriptor.java

```diff
@@ -122,48 +263,19 @@ Object callField(Object obj, StarlarkSemantics semantics, @Nullable Mutability m
    *
    * <p>The Mutability is used if it is necessary to allocate a Starlark copy of a Java result.
    */
-  Object call(Object obj, Object[] args, @Nullable Mutability mu)
+  Object call(
+      Object obj, @Nullable StarlarkThread starlarkThread, Object[] args, @Nullable Mutability mu)
       throws EvalException, InterruptedException {
     Preconditions.checkNotNull(obj);
-    Object result;
     try {
-      result = method.invoke(obj, args);
-    } catch (IllegalAccessException ex) {
-      // The annotated processor ensures that annotated methods are accessible.
-      throw new IllegalStateException(ex);
-
-    } catch (InvocationTargetException ex) {
-      Throwable e = ex.getCause();
-      if (e == null) {
-        throw new IllegalStateException(e);
-      }
+      return methodHandle.invokeExact(obj, starlarkThread, mu, args);
+    } catch (RuntimeException | Error | EvalException | InterruptedException e) {
       // Don't intercept unchecked exceptions.
-      Throwables.throwIfUnchecked(e);
-      if (e instanceof EvalException) {
-        throw (EvalException) e;
-      } else if (e instanceof InterruptedException) {
-        throw (InterruptedException) e;
-      } else {
-        // All other checked exceptions (e.g. LabelSyntaxException) are reported to Starlark.
-        throw new EvalException(null, null, e);
-      }
+      throw e;
+    } catch (Throwable e) {
```

Comment:
catch (Exception ex) ?

Reply:
`MethodHandle.invokeExact` declares `throws Throwable` because it does not wrap exception unlike `Method.invoke`.